torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -102,24 +106,35 @@ def tensorify_python_scalars(
     tracer = fx.proxy.GraphAppendingTracer(graph)
     expr_to_sym_proxy: dict[sympy.Expr, MetaProxy] = {}
     expr_to_tensor_proxy: dict[sympy.Expr, MetaProxy] = {}
+    specialized_float_nodes: dict[fx.Node, float] = {}
+    deleted: set[fx.Node] = set()
 
     first_non_placeholder = None
     placeholders = set()
     for node in graph.nodes:
         if node.op != "placeholder":
+            first_none_placeholder = node
             break
         else:
             placeholders.add(node)
 
     Analysis = TensorReferenceAnalysis
 
+    def dce(graph: fx.Graph, deleted: set[fx.Node]) -> None:
```

Comment:
Not sure why this is implemented from scratch as opposed to using eliminate_dead_code with a custom predicate.

Reply:
Refactored in the next diff. As discussed in chat, custom predicate doesn't eliminate item calls as we need but the default predicate does: https://github.com/pytorch/pytorch/pull/138869